docs/user-guide/logging/overview.md

```diff
@@ -0,0 +1,107 @@
+
+However, the native functionality provided by a container engine or runtime is usually not enough for a complete logging solution. For example, if a container crashes, a pod is evicted, or a node dies, you'll usually still want to access your application's logs. As such, logs should have a separate storage and lifecycle independent of nodes, pods, or containers; this concept is called __cluster-level-logging__. Cluster-level logging requires a separate back-end to store, analyze, and query logs. Kubernetes provides no native storage solution for logs data, but you can integrate many existing logging solutions into your Kubernetes cluster.
+
+In this document, you can find:
+
+* A basic demonstration of logging in Kubernetes using the standard output stream
+* A detailed description of the node logging architecture in Kubernetes
+* Guidance for implementing cluster-level logging in Kubernetes
+
+In the last section the assumption is that logging back-end is present inside or outside of the cluster. If you're not interested in having cluster-level logging, you might still find the description how logs are stored and handled on the node to be useful.
+
+## Basic logging in Kubernetes
+
+In this section, you can see an example of basic logging in Kubernetes that outputs data to the standard output stream. This demonstration uses a [pod specification](/docs/user-guide/logging/counter-pod.yaml) with a container that writes some text to standard output once per second.
+
+{% include code.html language="yaml" file="counter-pod.yaml" %}
+
+To run this pod, use the following command:
+
+```shell
+$ kubectl create -f counter-pod.yaml
+pod "counter" created
+```
+
+To fetch the logs, use the `kubectl logs` command, as follows
+
+```shell
+$ kubectl logs counter
+0: Tue Jun  2 21:37:31 UTC 2015
+1: Tue Jun  2 21:37:32 UTC 2015
+2: Tue Jun  2 21:37:33 UTC 2015
+3: Tue Jun  2 21:37:34 UTC 2015
+4: Tue Jun  2 21:37:35 UTC 2015
+5: Tue Jun  2 21:37:36 UTC 2015
+...
+```
+
+You can use `kubectl logs` to retrieve logs from a previous instantiation of a container with `--previous` flag, in case the container has crashed. If your pod has multiple containers, you should specify which container's logs you want to access by appending a container name to the command. See the [`kubectl logs` documentation](/docs/user-guide/kubectl/kubectl_logs) for more details.
+
+## Logging at the node level
+
+![Node level logging](/images/docs/user-guide/logging/logging-node-level.png)
+
+Everything a containerized application writes to `stdout` and `stderr` goes to a container runtime. Currently, in the Docker runtime, that output is written to a file using the standard [json logging driver](https://docs.docker.com/engine/admin/logging/overview).
```

Comment:
What I tried to say is that stdout and stderr stream are captured and redirected somewhere by the container engine, whether it's stdout and stderr of invoking process, journald, logging driver or anything else. The comment about working with each line separately is only about the current docker implementation, thanks for pointing that out.

We planned to write another doc for logging solution providers (e.g. ES, Datadoc, Splunk) to create their logging agent. This doc will containe the information about log paths, formats and maybe some other technical details to take into account. However this guide I wanted to make more user-oriented, without specifying all details. Makes sense?

WDYT about the following rephrasing:

Everything a containerized application writes to `stdout` and `stderr` is handled and redirected somewhere by a container engine. Currently, Docker container engine redirects those two streams to [a logging driver](https://docs.docker.com/engine/admin/logging/overview), which is configured in kubernetes to write to a file in json format.

**Note:** Docker json logging driver treats each line as a separate message. When it's used, there is no direct support for multi-line messages out of the box, it should be handled on the logging agent level or higher.